lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp

```diff
@@ -136,6 +136,37 @@ AreFuncletsOfSameAsyncClosure(swift::Demangle::NodePointer closure1,
   return Node::deepEquals(closure1_function, closure2_function);
 }
 
+std::optional<uint64_t>
+SwiftLanguageRuntime::GetFuncletNumber(swift::Demangle::NodePointer node) {
+  if (!node)
+    return {};
+  NodePointer node_id =
+      childAtPath(node, {Node::Kind::AsyncSuspendResumePartialFunction,
+                         Node::Kind::Number});
+  if (!node_id)
+    node_id = childAtPath(node, {Node::Kind::AsyncAwaitResumePartialFunction,
+                                 Node::Kind::Number});
+  if (node_id)
+    return node_id->getIndex();
```

Comment:
since I've seen all sorts of invariants break with demangled node trees, this should probably include a check to `node_id->hasIndex()`.